pkg/destroy/aws/aws.go

```diff
@@ -185,19 +185,23 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) ([]string, erro
 	// running on the cluster creating new resources while we are attempting to delete resources, which could leak
 	// the new resources.
 	ec2Client := ec2.New(awsSession)
+	timer := time.Now()
 	err = wait.PollImmediateUntil(
 		time.Second*10,
 		func() (done bool, err error) {
-			instancesToDelete, err := o.findEC2Instances(ctx, ec2Client, deleted)
+			instancesToDelete, instancesShuttingDown, err := o.findEC2Instances(ctx, ec2Client, deleted)
 			if err != nil {
 				o.Logger.WithError(err).Info("error while finding EC2 instances to delete")
 				if err := ctx.Err(); err != nil {
 					return false, err
 				}
 			}
-			if len(instancesToDelete) == 0 && err == nil {
+			if len(instancesToDelete) == 0 && len(instancesShuttingDown) == 0 && err == nil {
 				return true, nil
 			}
+			if time.Now().After(timer.Add(5 * time.Second)) {
+				instancesToDelete = append(instancesToDelete, instancesShuttingDown...)
```

Comment:
Doesn't the timer need to be refreshed here? Otherwise it will repeat every 10 seconds after the initial timer expires?